### DISCUSSION TITLE
Code signing and purchasing / trialling / why?

### DISCUSSION THREAD

**Discussion by ghost** [amidaware/tacticalrmm#1446] [General]
Apologies in advance if this upsets some people.

If I purchase code signing for my macOS endpoints and then end the "sponsorship" after a month or two will those endpoints stop reporting in? 

Given that there seems to be no way of even testing a macOS agent without paying maybe your should offer a 1 week trial code?

Given that most people will push the clients out via script, SSH etc the signing requirement seems a bit of a moot point to me. 

I understand that you need to make money to keep the project afloat (goodwill doesn't pay the bills) and voluntary donations didn't work for you so why not lower the price to something more sensible for code signing and enforce it for everyone after a trial period?

Surely code signing doesn't cost you $600 per year per user?

**Comment by wh1te909** [comment]
no they won't stop checking in, we're not assholes like that. the source code for the agent is public. you can compile your own mac agents without paying. or you can pay for code signing and get pre-compiled agents. you either do it yourself or you pay someone to do it for you, just like everything else in life. and code signing requires a legitimate legal entity which costs thousands of dollars a year in fees, taxes etc in addition to the cost for code signing.

**Reply by ghost** [reply]
Sponsored now :)

Quick question - to update the agent is it simply a case of running the command but with `update` instead of `install` ?

**Reply by wh1te909** [reply]
agents will update automatically and the updates are initiated server side, not client side
